src/material/tabs/tab-nav-bar/tab-nav-bar.ts

```diff
@@ -274,6 +298,42 @@ export class _MatTabLinkBase
     // have to update the focused index whenever the link receives focus.
     this._tabNavBar.focusIndex = this._tabNavBar._items.toArray().indexOf(this);
   }
+
+  _handleKeydown(event: KeyboardEvent) {
+    if (this._tabNavBar.tabPanel && event.keyCode === SPACE) {
+      this.elementRef.nativeElement.click();
+    }
+  }
+
+  _getAriaControls(): string | null {
+    return this._tabNavBar.tabPanel
+      ? this._tabNavBar.tabPanel?.id
+      : this.elementRef.nativeElement.getAttribute('aria-controls');
+  }
+
+  _getAriaSelected(): string | null {
+    if (this._tabNavBar.tabPanel) {
+      return this.active ? 'true' : 'false';
+    } else {
+      return this.elementRef.nativeElement.getAttribute('aria-selected');
```

Comment:
Instead of calling `getAttribute` for these every time, I think that we can inject them using `@Attribute` in the constructor. I _think_ that our binding will override any pre-existing binding anyway.

Reply:
Originally I tried to use `@Attribute`. The problem is that if someone adds an attribute template binding like `[attr.role]="role"` (as opposed to role="tablist"), then the `@Attribute` binding is actually `null`, because presumably the attribute template binding evaluates too late. That means we can't figure out what `role` they've added without using `getAttribute`. Here's a Stackblitz: https://stackblitz.com/edit/angular-ivy-d2ehto?file=src/app/app.component.ts.

Reply:
Hmm, interesting. I would've assumed that `getAttribute` will eventually start to return `null` because we clear it in our host bindings.

Reply:
It's not clear to me what you mean here. Can we mark this as resolved and use `getAttribute` or talk more over DM?

Reply:
Yeah, let's resolve it. I'm just surprised that it works.